### PR TITLE
Support hexadecimal string literals

### PR DESCRIPTION
**Pull request by benesch**

### CODE REVIEW

File: src/sqlparser.rs [hunk 1]
```diff
@@ -1037,6 +1040,16 @@ impl Parser {
                 Token::NationalStringLiteral(ref s) => {
                     Ok(Value::NationalStringLiteral(s.to_string()))
                 }
+                Token::HexStringLiteral(ref s) => {
+                    if let Some(bad) = s.chars().find(|c| !c.is_ascii_hexdigit()) {
```
Comment (nickolay) [inline src/sqlparser.rs]: [The spec](https://jakewheat.github.io/sql-overview/sql-2011-foundation-grammar.html#binary-string-literal) is more lenient, allowing whitespace as well, as far as I can see. Perhaps leave validation to the consumers?

File: src/sqltokenizer.rs [hunk 2]
```diff
@@ -286,6 +289,21 @@ impl<'a> Tokenizer<'a> {
                         }
                     }
                 }
+                x @ 'x' | x @ 'X' => {
```
Comment (nickolay) [inline src/sqltokenizer.rs]: Which dialect supports the lower-case 'x'?

File: src/sqltokenizer.rs [hunk 3]
```diff
@@ -286,6 +289,21 @@ impl<'a> Tokenizer<'a> {
...
+                    chars.next(); // consume, to check the next char
+                    match chars.peek() {
+                        Some('\'') => {
+                            // X'...' - a <national character string literal>
```
Comment (nickolay) [inline src/sqltokenizer.rs]: `<binary string literal>`

File: src/sqltokenizer.rs (hunk 2 above)
Comment (benesch) [reply src/sqltokenizer.rs]: PostgreSQL, at least. I've added a comment to this effect.

File: src/sqltokenizer.rs (hunk 3 above)
Comment (benesch) [reply src/sqltokenizer.rs]: Good catch, thanks.

File: src/sqlparser.rs (hunk 1 above)
Comment (benesch) [reply src/sqlparser.rs]: Sure thing. Done.

**Pull request merged**